### PR TITLE
core: Eigen Exp ExtractChannels

### PR DESCRIPTION
**Pull request by berak**

### CODE REVIEW

**Comment by berak** [comment]
tests stall on EigenNonSymmetric, looks like an opencv bug to me ;)

**Comment by deadprogram** [comment]
Hello @berak this PR has the same merge conflict from changes to `core_test.go`. Thanks again!

**Comment by berak** [comment]
hehe, finally found the button, conflict solved ;)

**Comment by berak** [comment]
conflicts resolved

**Comment by berak** [comment]
apologies, i was simply going too fast here, since merging one pr will invoke conflicts on any other.

(learned something, i hope ..)

File: core.go [hunk 1]
```diff
@@ -555,6 +556,63 @@ func CountNonZero(src Mat) int {
...
+	DctInverse                = DftInverse
+	DctRows                   = DftRows
+)
+
+// Dct Performs a forward or inverse discrete Cosine transform of 1D or 2D array.
+//
+// For further details, please see:
+// https://docs.opencv.org/3.4.0/d2/de8/group__core__array.html#ga85aad4d668c01fbd64825f589e3696d4
+//
+func Dct(src Mat, dst Mat, flags int) {
```
Comment (deadprogram) [inline core.go]: Hi @berak did you mean to make `flags` of type `DftFlags`? Seems like that was your intention here, which seems like a good thing.

Comment (berak) [reply core.go]: oh, ofc. 

mind, if i change Dft as well ?

Comment (deadprogram) [reply core.go]: Please do!

File: core.go [hunk 2]
```diff
@@ -555,6 +557,82 @@ func CountNonZero(src Mat) int {
...
+// ConvertScaleAbs Scales, calculates absolute values, and converts the result to 8-bit.
+//
+// For further details, please see:
+// https://docs.opencv.org/3.4.0/d2/de8/group__core__array.html#ga3460e9c9f37b563ab9dd550c4d8c4e7d
+//
+func ConvertScaleAbs(src Mat, dst Mat, alpha float64, beta float64) {
+	C.Mat_ConvertScaleAbs(src.p, dst.p, C.double(alpha), C.double(beta))
+}
+
+// CopyMakeBorder Forms a border around an image.
```
Comment (willdurand) [inline core.go]: I think we should have real English sentences in doc blocks, so I would not add uppercase to the verb.

File: core.go [hunk 3]
```diff
@@ -555,6 +557,82 @@ func CountNonZero(src Mat) int {
...
+		val4: C.double(value.A),
+	}
+
+	C.Mat_CopyMakeBorder(src.p, dst.p, C.int(top), C.int(bottom), C.int(left), C.int(right), C.int(borderType), c_value)
+}
+
+type DftFlags int
+
+const (
+	DftForward DftFlags = 0
```
Comment (willdurand) [inline core.go]: We should have doc for the flags, that would be very useful.

File: core.go [hunk 4]
```diff
@@ -555,6 +557,82 @@ func CountNonZero(src Mat) int {
...
+}
+
+// CopyMakeBorder Forms a border around an image.
+//
+// For further details, please see:
+// https://docs.opencv.org/3.4.0/d2/de8/group__core__array.html#ga2ac1049c2c3dd25c2b41bffe17658a36
+//
+func CopyMakeBorder(src Mat, dst Mat, top int, bottom int, left int, right int, borderType int, value color.RGBA) {
+
+	c_value := C.struct_Scalar{
```
Comment (willdurand) [inline core.go]: Shouldn't this be `cValue`? I am asking because in some other places, that was the case.

File: core.go (hunk 3 above)
Comment (berak) [reply core.go]: indeed.

File: core.go (hunk 2 above)
Comment (berak) [reply core.go]: right.

File: core.go (hunk 4 above)
Comment (berak) [reply core.go]: an implementation detail, but ok....

**Comment by deadprogram** [comment]
Thank you, now merging.

**Comment by berak** [comment]
thanks ;)

**Pull request merged**